# Data Analysis on Roller Coasters Dataset

In [2]:
# Import necessary libraries
import pandas as pd
import seaborn as sns
coasters = pd.read_csv("/coasters.csv") # load data

## What variables are available in this dataset?
'state', 'country', 'steel', 'construction', 'height', 'speed', 'length', 'inversions', 'numinversions', 'duration', 'gforce', 'opened', 'region', 'continent'

In [7]:
column_names = coasters.columns # get column names
column_names

Index(['state', 'country', 'steel', 'construction', 'height', 'speed',
       'length', 'inversions', 'numinversions', 'duration', 'gforce', 'opened',
       'region', 'continent'],
      dtype='object')

## How many coasters are there data for?
408

In [ ]:
len(coasters) # get length (i.e. number of rows) of dataset

408

## Do any of the descriptive statistics make any of the variables stick out from the rest?
Yes, the 'length' and 'numinversions' variables have standard deviations close to their means which may indicate high variation in these columns.

In [14]:
stats = coasters.describe() # get summary statistics for numerical variables
stats

steel      height       speed       length  numinversions  \
count  408.000000  326.000000  270.000000   318.000000     408.000000   
mean     0.897059   23.124871   69.362667   597.040716       0.784314   
std      0.304255   18.523768   29.327743   432.256987       1.664058   
min      0.000000    2.438400    9.720000    12.192000       0.000000   
25%      1.000000    8.650800   45.000000   291.000000       0.000000   
50%      1.000000   18.288000   68.850000   415.747500       0.000000   
75%      1.000000   33.167400   88.950000   833.121000       0.000000   
max      1.000000  128.016000  194.400000  2243.020000      10.000000   

         duration     gforce       opened      region  
count  192.000000  60.000000   380.000000  408.000000  
mean   112.462500   4.115000  1995.434211    1.786765  
std     50.518467   1.006896    12.753121    0.887323  
min      0.300000   2.100000  1924.000000    1.000000  
25%     75.000000   3.175000  1991.000000    1.000000  
50%    108.000000   4.500000  1999.000000    1.000000  
75%    140.750000   5.000000  2004.000000    3.000000  
max    300.000000   6.200000  2014.000000    3.000000

## Is standardization necessary for these variables?
Depending on the intended model used, standardization may be useful for this data, although it may not be necessary. However, a few times it would be necessary include, but are not limited to, when the model intended to be used assumes a normal distribution, is sensitive to variation, or when the variables are all on different scales. Since this data is not normalized, some of the variables have large coefficients of variation, and they are all on different scales, standardization would be necessary if the intended model met any of the above criteria.

In [15]:
coast_std = stats.loc['std'] # get standard deviation from descriptive statistics
coast_mean = stats.loc['mean'] # get mean from descriptive statistics
coast_cv = coast_std / coast_mean # calculate coefficent of variation for each variable
coast_cv

steel            0.339170
height           0.801032
speed            0.422817
length           0.723999
numinversions    2.121674
duration         0.449203
gforce           0.244689
opened           0.006391
region           0.496609
dtype: float64

## Are there any missing values?
Yes, there are missing values in the 'height', 'speed', 'length', 'duration', 'gforce', and 'opened' columns with the most missing values coming from the 'gforce' column.

In [16]:
coasters.isnull().sum() # get number of missing values for each column

state              0
country            0
steel              0
construction       0
height            82
speed            138
length            90
inversions         0
numinversions      0
duration         216
gforce           348
opened            28
region             0
continent          0
dtype: int64

## What continents are represented?
North America, Latin America, and Europe

In [17]:
unique_continents = coasters['continent'].unique() # get unique values of 'continent' variable
unique_continents

array(['North America', 'Latin America', 'Europe'], dtype=object)

## What is the tallest coaster included? fastest? longest?
Tallest: Coaster No. 193 at a height of 128.016

Fastest: Coaster No. 193 at a speed of 194.4

Longest: Coaster No. 164 at a length of 2243.02

In [19]:
tallest_index = coasters['height'].idxmax() # get index of tallest coaster
tallest_height = coasters.loc[tallest_index].height # get height of tallest coaster
[tallest_index, tallest_height]

[193, 128.016]

In [20]:
fastest_index = coasters['speed'].idxmax() # get index of fastest coaster
fastest_speed = coasters.loc[fastest_index].speed # get speed of fastest coaster
[fastest_index, fastest_speed]

[193, 194.4]

In [22]:
longest_index = coasters['length'].idxmax() # get index of longest coaster
longest_length = coasters.loc[longest_index].length # get length of longest coaster
[longest_index, longest_length]

[164, 2243.02]

## What is the average year coasters opened in North America? Latin America? Europe?
North America: 1992.56

Latin America: 2000.10

Europe: 1997.77

In [ ]:
avg_opened = coasters.groupby('continent')['opened'].mean() # get mean of 'opened' for each continent
avg_opened

continent
Europe           1997.774194
Latin America    2000.101695
North America    1992.563452
Name: opened, dtype: float64

## How does the average height, speed, and length change based on if the coasters is wood or steel?

Wood coasters are on average taller, faster, and longer than steel ones.

In [ ]:
height_construct = coasters.groupby('construction')['height'].mean() # get mean of 'height' for each construction
height_construct

construction
Steel    22.082011
Wood     31.028653
Name: height, dtype: float64

In [ ]:
speed_construct = coasters.groupby('construction')['speed'].mean() # get mean of 'speed' for each construction
speed_construct

construction
Steel    66.468475
Wood     89.451765
Name: speed, dtype: float64

In [ ]:
length_construct = coasters.groupby('construction')['length'].mean() # get mean of 'length' for each construction
length_construct

construction
Steel     528.607143
Wood     1072.654050
Name: length, dtype: float64

## Which continent has the tallest, fastest, and/or longest average coaster?

North America has the tallest, fastest, and longest average coaster.

In [ ]:
tallest_continent = coasters.groupby('continent')['height'].mean() # get mean of 'height' for each continent
tallest_continent

continent
Europe           16.028448
Latin America    21.327692
North America    29.607641
Name: height, dtype: float64

In [ ]:
fastest_continent = coasters.groupby('continent')['speed'].mean() # get mean of 'speed' for each continent
fastest_continent

continent
Europe           54.955789
Latin America    67.171429
North America    81.077143
Name: speed, dtype: float64

In [ ]:
longest_continent = coasters.groupby('continent')['length'].mean() # get mean of 'length' for each continent
longest_continent

continent
Europe           484.440000
Latin America    493.067692
North America    722.590188
Name: length, dtype: float64